projects/plugins/jetpack/extensions/blocks/tiled-gallery/tiled-gallery.php

```diff
@@ -210,6 +213,602 @@ private static function is_squareish_layout( $attr ) {
+					$id = ! empty( $image_data['id'] ) ? absint( $image_data['id'] ) : 0;
+
+					$images[] = array(
+						'url' => $url,
+						'alt' => $alt_text,
+						'id'  => $id,
+					);
+				}
+			}
+		}
+
+		return $images;
+	}
+
+	/**
+	 * Get layout style and columns from block attributes.
+	 *
+	 * @param array $attr Block attributes.
+	 * @return array Array with 'style', 'columns', and 'border_radius' keys.
+	 */
+	private static function get_layout_style_from_attributes( $attr ) {
+		$layout_info = array(
+			'style'         => 'rectangular', // Default to rectangular/mosaic layout
+			'columns'       => 3, // Default to 3 columns
+			'border_radius' => 0, // Default to no border radius
+		);
+
+		// Get number of columns from attributes with validation
+		if ( ! empty( $attr['columns'] ) && is_numeric( $attr['columns'] ) ) {
+			$columns = absint( $attr['columns'] );
+			// Clamp columns between 1 and 6 for reasonable layouts
+			$layout_info['columns'] = max( 1, min( 6, $columns ) );
+		}
+
+		// Get border radius from roundedCorners attribute (preferred method)
+		if ( ! empty( $attr['roundedCorners'] ) && is_numeric( $attr['roundedCorners'] ) ) {
+			$border_radius_value = absint( $attr['roundedCorners'] );
+			// Clamp value between 0 and 20
+			$layout_info['border_radius'] = max( 0, min( 20, $border_radius_value ) );
+		}
+
+		// Get layout style and border radius from className
+		if ( ! empty( $attr['className'] ) ) {
+			if ( str_contains( $attr['className'], 'is-style-square' ) ) {
+				$layout_info['style'] = 'square';
+			} elseif ( str_contains( $attr['className'], 'is-style-circle' ) ) {
+				$layout_info['style'] = 'circle';
+			} elseif ( str_contains( $attr['className'], 'is-style-columns' ) ) {
+				$layout_info['style'] = 'columns';
+			}
+
+			// Extract border radius from has-rounded-corners-{value} class (fallback method)
+			if ( $layout_info['border_radius'] === 0 && preg_match( '/has-rounded-corners-(\d+)/', $attr['className'], $matches ) ) {
+				$border_radius_value = absint( $matches[1] );
+					// Clamp value between 0 and 20
+				$layout_info['border_radius'] = max( 0, min( 20, $border_radius_value ) );
+			}
+		}
+
+		return $layout_info;
+	}
+
+	/**
+	 * Build email layout content based on layout style and columns.
+	 *
+	 * @param array $images Array of image data.
+	 * @param array $layout_info Array with 'style' and 'columns' keys.
+	 * @param int   $target_width Target width for email.
+	 * @param int   $cell_padding Cell padding.
+	 * @return string HTML content.
+	 */
+	private static function build_email_layout_content( $images, $layout_info, $target_width, $cell_padding ) {
+		$layout_style  = $layout_info['style'];
+		$columns       = $layout_info['columns'];
+		$border_radius = $layout_info['border_radius'];
+
+		switch ( $layout_style ) {
+			case 'square':
+				return self::build_square_layout_content( $images, $target_width, $cell_padding, $columns, 'square', $border_radius );
```

Comment:
The `$target_width` doesn't seem to be used in any of the individual `build_*_layout_content` methods, so I think it could be removed.